pkg/executor/infoschema_reader.go

```diff
@@ -781,11 +782,16 @@ func (e *hugeMemTableRetriever) setDataForColumns(ctx context.Context, sctx sess
 	e.rows = e.rows[:0]
 	batch := 1024
 	is := sctx.GetInfoSchema().(infoschema.InfoSchema)
-	for ; e.dbsIdx < len(e.dbs); e.dbsIdx++ {
+	for e.dbsIdx < len(e.dbs) {
 		schema := e.dbs[e.dbsIdx]
-		tables := is.SchemaTables(schema)
-		for e.tblIdx < len(tables) {
-			table := tables[e.tblIdx]
+		if len(e.tables) == 0 {
+			e.tables = is.SchemaTables(schema)
+			sort.Slice(e.tables, func(i, j int) bool {
```

Comment:
slices.SortFunc will be better. It can avoid unnecessary bounds check elimination.

Reply:
The sort is really slow, I prefer to revert change from https://github.com/pingcap/tidb/pull/51436 for now to avoid regression